src/main/java/gameframework/motion/MoveStrategyStraightLine.java

```diff
@@ -1,17 +1,31 @@
 package gameframework.motion;
 
 import java.awt.Point;
-
+/**
+ *  * {@link MoveStrategy} which select the direction to reach the goal point
+ *  {@link SpeedVector speed vectors} based on the goal point
+ *
+ */
 public class MoveStrategyStraightLine implements MoveStrategy {
 
 	protected int speed = 8;
 	Point goal, currentPosition;
 
+	/**
+	 * if no speed parameter, the default speed is set to 8
```

Comment:
Please fix the code duplication instead of documenting it… delegate to the canonical constructor and pass the default value for the missing parameter…